### ISSUE TITLE
I cannot open a remote repository

### ISSUE THREAD

**Issue by jrwblackwell** [microsoft/vscode-remote-repositories-github#332]
Type: <b>Bug</b>

I installed the extension

Extension version: 0.32.0
VS Code version: Code 1.75.0 (Universal) (e2816fe719a4026ffa1ee0189dc89bdfdbafb164, 2023-02-01T15:24:42.903Z)
OS version: Darwin arm64 22.2.0
Modes:
Sandboxed: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 (8 x 24)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|4, 4, 3|
|Memory (System)|8.00GB (0.06GB free)|
|Process Argv|--crash-reporter-id 3cdc581a-e2a1-4444-aa78-6a6f96f83fac|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes627:30244334
vslsvsres303:30308271
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263cf:30335440
pythondataviewer:30285071
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593:30376534
pythonvs932:30410667
cppdebug:30492333
vsclangdf:30486550
c4g48928:30535728
dsvsc012:30540252
pynewext54:30695312
azure-dev_surveyone:30548225
vscccc:30610679
3biah626:30602489
pyind779:30671433
89544117:30613380
pythonsymbol12:30671437
2i9eh265:30646982
showlangstatbar:30737416
azdwalk:30721579
pythonms35:30701012
pythonfmttext:30731395
pythoncmvfstr:30731969
fixhidewlkth:30730051

```

</details>

**Comment by joyceerhl** [comment]
@jrwblackwell apologies for missing this, do you have the GitHub Repositories extension installed and enabled? https://marketplace.visualstudio.com/items?itemName=GitHub.remotehub

**Comment by xmedeko** [comment]
Do not know if it's the same bug, but cannot open remote repository from https://vscode.dev/ (Firefox, Chrome). I enter GitHub URL of my repo, then confirm GitHub OAuth screen and then nothing happens.

In dev console, I see:
```
Error: command '_workbench.getTarballProxyEndpoints' not found
Error: command '_remotehub.getLfsProxyEndpoint' not found
```

GitHub Repositories v0.60.0

**Comment by joyceerhl** [comment]
@xmedeko I am unable to reproduce the behavior you describe, can you please provide a screen recording or the URL of the repo you entered?

**Comment by xmedeko** [comment]
1. Having Firefox, signed in GitHub already.
2. Opened https://vscode.dev/ from Open Remote Repository.
3. Entered https://github.com/xmedeko/install.sh (I have tried a few other my repositories, too.)
4. Dialog popped up: "The extension 'GitHub Repositories' wants to sign in using GitHub.", Clicked "Allow".
5. In the bottom right corner notification "Signing in to github.com ..." ~~goes forever.~~ _Update:_ notification is displayed for 5 minutes. 

In the DevTools console, I see errors I have mentioned above.

Note: When I am not signed in to GitHub, the problem is the same. Just I have to sign in by GitHub OAuth screen in the step 4.

Note2: I have tried do delete all cookies and data of page `https://vscode.dev/` - still the same problem.

**Comment by joyceerhl** [comment]
Do you have a network firewall that might be blocking the auth process? cc @TylerLeonhardt

**Comment by TylerLeonhardt** [comment]
Can you follow these steps so that we can get more information on what's going on?

* Trigger the sign in
* _Wait for the issue to happen..._
* Open the Output pane (cmd+shift+u I think on macOS)
* drop in this issue he contents of the following:
  * `GitHub Authentication` Output channel

**Comment by xmedeko** [comment]
Nope, no proxy. GitHub OAuth (Google OAuth and others, too) works for other sites well.

## Firefox 
**Update:** "Signing in to github.com ..." does not go forever, just for 5 minutes. Then timeout error is shown.

@TylerLeonhardt `GitHub Authentication` Output channel (FF, when I am signed to GitHub already):
```
2023-08-08 08:17:22.729 [info] Reading sessions from keychain...
2023-08-08 08:17:22.729 [info] Getting sessions for all scopes...
2023-08-08 08:17:22.729 [info] Got 0 sessions for ...
2023-08-08 08:17:22.729 [info] Getting sessions for user:email...
2023-08-08 08:17:22.729 [info] Got 0 sessions for user:email...
2023-08-08 08:17:22.729 [info] Getting sessions for all scopes...
2023-08-08 08:17:22.729 [info] Got 0 sessions for ...
2023-08-08 08:17:22.843 [info] Getting sessions for all scopes...
2023-08-08 08:17:22.843 [info] Got 0 sessions for ...
2023-08-08 08:17:35.816 [info] Getting sessions for read:user,repo,user:email,workflow...
2023-08-08 08:17:35.816 [info] Got 0 sessions for read:user,repo,user:email,workflow...
2023-08-08 08:17:49.834 [info] Getting sessions for read:user,repo,user:email,workflow...
2023-08-08 08:17:49.834 [info] Got 0 sessions for read:user,repo,user:email,workflow...
2023-08-08 08:17:52.337 [info] Logging in for the following scopes: read:user repo user:email workflow
2023-08-08 08:17:52.369 [info] Trying without local server... (read:user repo user:email workflow)
2023-08-08 08:22:52.362 [error] Timed out
2023-08-08 08:22:52.368 [error] login@https://main.vscode-cdn.net/stable/6445d93c81ebe42c4cbd7a60712e0b17d9463e97/extensions/github-authentication/dist/browser/extension.js#vscode-extension:4:187979

2023-08-08 08:22:52.382 [info] Getting sessions for read:user,repo,user:email,workflow...
2023-08-08 08:22:52.385 [info] Got 0 sessions for read:user,repo,user:email,workflow...
```

In DevTools, I still see the errors (some of them mentioned in https://github.com/microsoft/vscode-remote-repositories-github/issues/332#issuecomment-1649498231 already):
```
INFO Found additional builtin extension gallery resources in env ...
INFO Updating additional builtin extensions cache
WARN Timed out waiting for all supported executions
unreachable code after return statement
Error: command '_workbench.getTarballProxyEndpoints' not found
Error: command '_remotehub.getLfsProxyEndpoint' not found
Sign in failed: Error: No auth flow succeeded
ERR No provider registered with : Zt@https://ms-vscode.vscode-unpkg.net/ms-vscode/remote-repositories/0.38.1/extension/dist/browser/extension.js#vscode-extension:53:2757 ...
```

_Note 1:_ Same issue when I use (user icon in the left bottom corner) _Accounts > Sign In with GitHub to use GitHub Repositories (1)_. A bit faster to trigger this issue.

## Chrome Works Now
_Note 2:_ It works from Chrome (115, just upgraded) now, it has not been working from Chrome 2 weeks ago. 2 weeks ago, I have seen in Chrome the `command ... not found` errors mentioned in https://github.com/microsoft/vscode-remote-repositories-github/issues/332#issuecomment-1649498231 **But do not see them now.** I see in DevTools console logs:
```
INFO Migrating workspace extension storage from github.remotehub to GitHub.remotehub...
INFO Migrated workspace extension storage from github.remotehub to GitHub.remotehub
```

**Comment by TylerLeonhardt** [comment]
Bizarre, so for FF can you [attach a gif](https://gifcap.dev/) or video of you going through the auth flow? I'm curious at what part the flow gets stuck because from the logs VS Code is saying "the flow never came back to us from the browser".

**Comment by xmedeko** [comment]
@TylerLeonhardt I have found the problem already, I had blocked popup windows for vscode.dev without notice - i.e. Firefox blocks popups without any message. (Do not know why, maybe I have chosen that option by a mistake in the past. Strange thing is, that I had similar problem in Chrome, but I use Chrome seldom - just for pages which have do not work well in FF). After unblocking popups, it works well. Sorry for bothering you.

Also I have noted that errors in DevTools console does not appear after unblocking popups, including the messages: 
```
WARN Timed out waiting for all supported executions
unreachable code after return statement
```
etc.

Tip: In case of timeout error from OAuth the error message notice in VS Code could mention to unblock popups.
